README.md

```diff
@@ -2,20 +2,136 @@
 
 > Test client for Xo-Server
 
-## Usage
-
-xo-server needs to be installed on `localhost:9000`.
-You should also have a `admin@admin.net` account with password `admin`.
+You should have a `admin@admin.net` account with password `admin`.
 
-There should be a Xen Server available at `192.168.1.3`with the credentials `root` and password `qwerty`.
+## Adding a test
 
-Warning ! This client will create and delete a lot of items (users, groups, servers ...) to test features
+### Organization
 
 ```
-> npm ci
-> npm test
+src
+├─ user
+|   ├─ __snapshots__
+|   |     └─ index.spec.js.snap 
+|   └─ index.spec.js
+├─ job
+¦   └─ index.spec.js
+¦
+¦
+├─ _xoWithTestHelpers.js
+└─ util.js
+```
+    
+Let's take an example of `index.spec.js` from `user/`:
+```
+describe("user", () => {
+  describe(".create() :", () => {
+    it("creates a user with permission", async () => {
+      /* some tests */
+    });
+    it("creates a user without permission", async () => {
+      /* some tests */
+    });
+  });
+
+  describe(".delete() :", () => {
+    it("deletes a user successfully with id", async () => {
+      /* some tests */
+    });
+  });
+
+  it("creates two users, modifies a user email to be the same with the other and fails trying to connect them", async () => {
+    /* some tests */
+  });
+});
 ```
 
+### Test environment
+
+- The test environment must remain the same before and after each test:  
+  * each resource created must be deleted  
+  * existing resources should not be altered
+
+  There is a helper `_xoWithTestHelpers.js` to create temporary resources and to interface with XO.  
+  For example, use `xo.createTempUser(params)` instead of `xo.call("create.user", params)` to create user:
+  ```
+  describe(".getAll() :", () => {
+    it("gets all the users created", async () => {
+      const userId1 = await xo.createTempUser({
+        email: "wayne4@vates.fr",
+        password: "batman",
+        permission: "user",
+      });
+      const userId2 = await xo.createTempUser({
+        email: "wayne5@vates.fr",
+        password: "batman",
+        permission: "user",
+      });
+      let users = await xo.call("user.getAll");
+      expect(Array.isArray(users)).toBe(true);
+      users = keyBy(users, "id");
+      expect([
+        omit(users[userId1], "id"),
+        omit(users[userId2], "id"),
+      ]).toMatchSnapshot();
+    });
+    // The users will be deleted automatically at the end of the test
```

Comment:
This comment refer to the method `createTempUser`.  IMHO, it better to move it above it also rename `users` -> `user `.